mali: support building against 4.10

In 4.10, the virtual_address pointer in struct vma_fault is gone and
should be replaced by using the address field in the same structure.

See commit 1a29d85eb0f1 ("mm: use vmf->address instead of of
vmf->virtual_address")

Signed-off-by: Maxime Ripard <maxime.ripard@free-electrons.com>
--- a/src/devicedrv/mali/linux/mali_memory.c
+++ b/src/devicedrv/mali/linux/mali_memory.c
@@ -64,7 +64,12 @@ static int mali_mem_vma_fault(struct vm_
 	int ret;
 	int prefetch_num = MALI_VM_NUM_FAULT_PREFETCH;
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 10, 0)
+	unsigned long address = vmf->address;
+#else
 	unsigned long address = (unsigned long)vmf->virtual_address;
+#endif
+
 	MALI_DEBUG_ASSERT(alloc->backend_handle);
 	MALI_DEBUG_ASSERT((unsigned long)alloc->cpu_mapping.addr <= address);
 
